source/carton-runner-interface/src/do_not_modify/storage.rs

```diff
@@ -60,4 +60,24 @@ where
         let data = self.data.as_mut_ptr();
         unsafe { ndarray::ArrayViewMut::from_shape_ptr(self.get_shape(), data) }
     }
+
+    pub unsafe fn into_bytes(self) -> (Vec<u8>, Vec<u64>) {
+        // TODO: unsafe because not guaranteed to be numeric, maybe move elsewhere.
+        let len = self.strides.iter().product() as usize * std::mem::size_of::<T>();
+        let buffer = unsafe { Vec::from_raw_parts(
+            self.data.as_ptr() as *mut u8,
+            len,
+            len,
+        ) };
+        (buffer, self.shape)
+    }
+
+    pub fn into_vec(self) -> (Vec<T>, Vec<u64>) {
+        let len = self.strides.iter().product() as usize;
+        let buffer = unsafe { Vec::from_raw_parts(
+            self.data.as_ptr() as *mut T,
+            len,
+            len,
+        ) };
+    }
```

Comment:
> Also I added 2 methods to TensorStorage, are those alright?

Couple notes. The `len` calculation here is incorrect (you're using strides instead of shape).

Also `Vec::from_raw_parts` has a lot of [caveats](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.from_raw_parts). I wouldn't recommend using it if you don't control how the pointer was allocated.

Generally, I'd recommend using `.view().as_slice().unwrap().to_vec()` to convert from a Tensor to a `Vec`, That'll also panic if the memory ordering is different than expected rather than silently exposing the wrong data.

I'd suggest just writing a few conversion utilities in your crate (like in the [TorchScript runner](https://github.com/VivekPanyam/carton/blob/f208de72c85512c600a8f84997be9c9a4c74f18e/source/carton-runner-torch/src/main.rs#L138-L283)).

These methods don't seem broadly useful enough to go into the runner interface directly (especially since both of them can be easily done in a safer manner on top of `.view()`).